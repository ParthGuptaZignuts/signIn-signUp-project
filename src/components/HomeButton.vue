<script setup lang="ts"></script>

<template>
  <div class="demo-space-x">
    <VBtn variant="flat" color="info" to="/"> Home</VBtn>
  </div>
</template>
