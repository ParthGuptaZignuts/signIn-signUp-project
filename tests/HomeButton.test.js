import { mount } from '@vue/test-utils'
import HomeButton from "../src/components/HomeButton.vue"
jest.mock('vuetify/components', () => ({
  VBtn: {
    template: '<button><slot /></button>',
    props: ['variant', 'color', 'to']
  }
}))

describe('HomeButton.vue', () => {
  it('renders Home button with correct attributes', () => {
    const wrapper = mount(HomeButton)
    const button = wrapper.find('button')

    expect(button.exists()).toBe(true)
    expect(button.text()).toBe('Home')
    expect(button.attributes('variant')).toBe('flat')
    expect(button.attributes('color')).toBe('info')
    expect(button.attributes('to')).toBe('/')
  })

  it('has correct class and structure', () => {
    const wrapper = mount(HomeButton)
    const div = wrapper.find('div.demo-space-x')

    expect(div.exists()).toBe(true)
    expect(div.find('button').exists()).toBe(true)
  })

  it('matches the snapshot', () => {
    const wrapper = mount(HomeButton)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
