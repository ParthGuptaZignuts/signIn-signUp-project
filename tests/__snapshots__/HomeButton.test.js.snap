// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeButton.vue matches the snapshot 1`] = `
"<div class="demo-space-x">
  <vbtn variant="flat" color="info" to="/"> Home</vbtn>
</div>"
`;
